owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/files/RemoteFile.kt

```diff
@@ -0,0 +1,153 @@
+ *   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *   copies of the Software, and to permit persons to whom the Software is
+ *   furnished to do so, subject to the following conditions:
+ *
+ *   The above copyright notice and this permission notice shall be included in
+ *   all copies or substantial portions of the Software.
+ *
+ *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ *   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ *   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ *   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ *   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *   THE SOFTWARE.
+ *
+ */
+
+package com.owncloud.android.lib.resources.files
+
+import android.net.Uri
+import android.os.Parcelable
+import at.bitfire.dav4jvm.Response
+import at.bitfire.dav4jvm.property.CreationDate
+import at.bitfire.dav4jvm.property.GetContentLength
+import at.bitfire.dav4jvm.property.GetContentType
+import at.bitfire.dav4jvm.property.GetETag
+import at.bitfire.dav4jvm.property.GetLastModified
+import at.bitfire.dav4jvm.property.OCId
+import at.bitfire.dav4jvm.property.OCPermissions
+import at.bitfire.dav4jvm.property.OCPrivatelink
+import at.bitfire.dav4jvm.property.OCSize
+import at.bitfire.dav4jvm.property.QuotaAvailableBytes
+import at.bitfire.dav4jvm.property.QuotaUsedBytes
+import com.owncloud.android.lib.common.OwnCloudClient
+import kotlinx.android.parcel.Parcelize
+import okhttp3.HttpUrl
+import java.io.File
+import java.math.BigDecimal
+
+/**
+ * Contains the data of a Remote File from a WebDavEntry
+ *
+ * The path received must be URL-decoded. Path separator must be File.separator, and it must be the first character in 'path'.
+ *
+ * @author masensio
+ * @author Christian Schabesberger
+ * @author Abel García de Prada
+ */
+@Parcelize
+data class RemoteFile(
+    var remotePath: String,
+    var mimeType: String = "DIR",
+    var length: Long = 0,
+    var creationTimestamp: Long = 0,
+    var modifiedTimestamp: Long = 0,
+    var etag: String? = null,
+    var permissions: String? = null,
+    var remoteId: String? = null,
+    var size: Long = 0,
+    var quotaUsedBytes: BigDecimal? = null,
+    var quotaAvailableBytes: BigDecimal? = null,
+    var privateLink: String? = null,
+    var owner: String
+) : Parcelable {
+
+    // TODO: Quotas not used. Use or remove them.
+    init {
+        require(!(remotePath.isEmpty() || !remotePath.startsWith(File.separator))) { "Trying to create a OCFile with a non valid remote path: $remotePath" }
+    }
+
+    /**
+     * Use this to find out if this file is a folder.
+     *
+     * @return true if it is a folder
+     */
+    val isFolder
+        get() = mimeType == MIME_DIR || mimeType == MIME_DIR_UNIX
```

Comment:
I was wondering if there is a more elegant way to write this, you could create an extension for this scenario:

```
fun Any.isOneOf(vararg values: Any): Boolean {
    return this in values
}
```

And use it like

`mimeType.isOneOf(MIME_DIR, MIME_DIR_UNIX)`

It is more scalable and will be most likely used in other parts of the code, it would be great that Kotlin had something like this already 😩 

It's just a suggestion, add it if you want to.

Reply:
I like it! Added 🍻